### PR TITLE
Create and download diagnostics bundle.

### PR DESCRIPTION
**Pull request by jeschkies**
## High-level description

This adds the `diagnostics-%.zip` target which will create and download a diagnostics
bundle from a cluster.

## Corresponding DC/OS tickets (required)

- [D2IQ-65942](https://jira.d2iq.com/browse/D2IQ-65942) Create and download diagnostics bundle

### CODE REVIEW

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ready for Review

**Comment by jeschkies** [comment]
@mesosphere-mergebot request-review @janisz

**Comment by jeschkies** [comment]
@mesosphere-mergebot request-review @DominikDary

File: test_util/Makefile [hunk 1]
```diff
@@ -80,6 +86,39 @@ clean:
...
+	@echo "##teamcity[blockOpened name='diagnostics-bundle' description='Download Diagnostics bundle']"
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<) -- \
+		"curl -X PUT $(diagnostics_api)/$*"
+	touch $@
+
+# Wait until diagnostics bundle is ready.
+diagnostics-%.done: cluster.json diagnostics-%.started 
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<) -- \
+		"curl -s $(diagnostics_api)/$* -o status.json; \
+		until jq '.status' status.json | grep -q 'Done'; do cat status.json; curl -s $(diagnostics_api)/$* -o status.json; sleep 5; done;"
```
Comment (janisz) [inline test_util/Makefile]: How about moving loop outside of SSH to not keep open connection for a long time.

**Comment by DominikDary** [comment]
@mesosphere-mergebot changelog-not-required internal change

File: test_util/Makefile (hunk 1 above)
Comment (jeschkies) [reply test_util/Makefile]: I'm not really sure how to do that. Has `ssh` the exit code as the subcommand?

**Comment by jeschkies** [comment]
@mesosphere-mergebot changelog-not-required internal changes again

**Comment by jeschkies** [comment]
@mesosphere-mergebot sync

**Comment by jeschkies** [comment]
@mesosphere-mergebot sync

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ship It

**Pull request merged**